// @trinsic/service-clients/dist/provider/models/mappers.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudWalletAnalyticsContract = {
    serializedName: "CloudWalletAnalyticsContract",
    type: {
        name: "Composite",
        className: "CloudWalletAnalyticsContract",
        modelProperties: {
            total: {
                serializedName: "total",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
exports.UsageContract = {
    serializedName: "UsageContract",
    type: {
        name: "Composite",
        className: "UsageContract",
        modelProperties: {
            periodStart: {
                serializedName: "periodStart",
                type: {
                    name: "DateTime"
                }
            },
            periodEnd: {
                serializedName: "periodEnd",
                type: {
                    name: "DateTime"
                }
            },
            credentialCount: {
                serializedName: "credentialCount",
                type: {
                    name: "Number"
                }
            },
            verificationCount: {
                serializedName: "verificationCount",
                type: {
                    name: "Number"
                }
            },
            invoiceId: {
                serializedName: "invoiceId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.AssetContract = {
    serializedName: "AssetContract",
    type: {
        name: "Composite",
        className: "AssetContract",
        modelProperties: {
            publicUri: {
                serializedName: "publicUri",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.NetworkContract = {
    serializedName: "NetworkContract",
    type: {
        name: "Composite",
        className: "NetworkContract",
        modelProperties: {
            networkId: {
                serializedName: "networkId",
                type: {
                    name: "String"
                }
            },
            networkName: {
                serializedName: "networkName",
                type: {
                    name: "String"
                }
            },
            genesisTransactions: {
                serializedName: "genesisTransactions",
                type: {
                    name: "String"
                }
            },
            poolProtocolVersion: {
                serializedName: "poolProtocolVersion",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
exports.TenantExtendedInformationContract = {
    serializedName: "TenantExtendedInformationContract",
    type: {
        name: "Composite",
        className: "TenantExtendedInformationContract",
        modelProperties: {
            issuerDid: {
                serializedName: "issuerDid",
                type: {
                    name: "String"
                }
            },
            issuerKey: {
                serializedName: "issuerKey",
                type: {
                    name: "String"
                }
            },
            issuerKeyGenerationSeed: {
                serializedName: "issuerKeyGenerationSeed",
                type: {
                    name: "String"
                }
            },
            agentDid: {
                serializedName: "agentDid",
                type: {
                    name: "String"
                }
            },
            agentKey: {
                serializedName: "agentKey",
                type: {
                    name: "String"
                }
            },
            agentKeyGenerationSeed: {
                serializedName: "agentKeyGenerationSeed",
                type: {
                    name: "String"
                }
            },
            agentServiceEndpoint: {
                serializedName: "agentServiceEndpoint",
                type: {
                    name: "String"
                }
            },
            transactionEndorsement: {
                serializedName: "transactionEndorsement",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.TenantContract = {
    serializedName: "TenantContract",
    type: {
        name: "Composite",
        className: "TenantContract",
        modelProperties: {
            name: {
                readOnly: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            imageUrl: {
                serializedName: "imageUrl",
                type: {
                    name: "String"
                }
            },
            network: {
                serializedName: "network",
                type: {
                    name: "Composite",
                    className: "NetworkContract"
                }
            },
            tenantId: {
                readOnly: true,
                serializedName: "tenantId",
                type: {
                    name: "String"
                }
            },
            region: {
                serializedName: "region",
                type: {
                    name: "String"
                }
            },
            extendedInformation: {
                serializedName: "extendedInformation",
                type: {
                    name: "Composite",
                    className: "TenantExtendedInformationContract"
                }
            }
        }
    }
};
exports.TenantParameters = {
    serializedName: "TenantParameters",
    type: {
        name: "Composite",
        className: "TenantParameters",
        modelProperties: {
            tenantId: {
                serializedName: "tenantId",
                type: {
                    name: "String"
                }
            },
            issuerSeed: {
                serializedName: "issuerSeed",
                type: {
                    name: "String"
                }
            },
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            imageUrl: {
                serializedName: "imageUrl",
                type: {
                    name: "String"
                }
            },
            networkId: {
                serializedName: "networkId",
                type: {
                    name: "String"
                }
            },
            endorserType: {
                serializedName: "endorserType",
                type: {
                    name: "String"
                }
            },
            region: {
                serializedName: "region",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.TenantUpdateParameters = {
    serializedName: "TenantUpdateParameters",
    type: {
        name: "Composite",
        className: "TenantUpdateParameters",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            imageUrl: {
                serializedName: "imageUrl",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.TenantKeysContract = {
    serializedName: "TenantKeysContract",
    type: {
        name: "Composite",
        className: "TenantKeysContract",
        modelProperties: {
            accessToken: {
                serializedName: "accessToken",
                type: {
                    name: "String"
                }
            },
            subscriptionKey: {
                serializedName: "subscriptionKey",
                type: {
                    name: "String"
                }
            }
        }
    }
};
