// @trinsic/service-clients/dist/provider/models/parameters.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.contentType = {
    parameterPath: [
        "options",
        "contentType"
    ],
    mapper: {
        serializedName: "contentType",
        type: {
            name: "String"
        }
    }
};
exports.filename = {
    parameterPath: [
        "options",
        "filename"
    ],
    mapper: {
        serializedName: "filename",
        type: {
            name: "String"
        }
    }
};
exports.tenantId = {
    parameterPath: "tenantId",
    mapper: {
        required: true,
        serializedName: "tenantId",
        type: {
            name: "String"
        }
    }
};
exports.uploadedFiles = {
    parameterPath: "uploadedFiles",
    mapper: {
        required: true,
        serializedName: "uploadedFiles",
        type: {
            name: "Stream"
        }
    }
};
