// @trinsic/service-clients/dist/wallet/models/mappers.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentEndpoint = {
    serializedName: "AgentEndpoint",
    type: {
        name: "Composite",
        className: "AgentEndpoint",
        modelProperties: {
            did: {
                serializedName: "did",
                type: {
                    name: "String"
                }
            },
            verkey: {
                serializedName: "verkey",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            uri: {
                serializedName: "uri",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.ConnectionContract = {
    serializedName: "ConnectionContract",
    type: {
        name: "Composite",
        className: "ConnectionContract",
        modelProperties: {
            connectionId: {
                serializedName: "connectionId",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            imageUrl: {
                serializedName: "imageUrl",
                type: {
                    name: "String"
                }
            },
            myDid: {
                serializedName: "myDid",
                type: {
                    name: "String"
                }
            },
            theirDid: {
                serializedName: "theirDid",
                type: {
                    name: "String"
                }
            },
            myKey: {
                serializedName: "myKey",
                type: {
                    name: "String"
                }
            },
            theirKey: {
                serializedName: "theirKey",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            invitation: {
                serializedName: "invitation",
                type: {
                    name: "String"
                }
            },
            invitationUrl: {
                serializedName: "invitationUrl",
                type: {
                    name: "String"
                }
            },
            endpoint: {
                serializedName: "endpoint",
                type: {
                    name: "Composite",
                    className: "AgentEndpoint"
                }
            },
            createdAtUtc: {
                serializedName: "createdAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            multiParty: {
                serializedName: "multiParty",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.CredentialContract = {
    serializedName: "CredentialContract",
    type: {
        name: "Composite",
        className: "CredentialContract",
        modelProperties: {
            credentialId: {
                serializedName: "credentialId",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            connectionId: {
                serializedName: "connectionId",
                type: {
                    name: "String"
                }
            },
            definitionId: {
                serializedName: "definitionId",
                type: {
                    name: "String"
                }
            },
            schemaId: {
                serializedName: "schemaId",
                type: {
                    name: "String"
                }
            },
            offerData: {
                serializedName: "offerData",
                type: {
                    name: "String"
                }
            },
            offerUrl: {
                serializedName: "offerUrl",
                type: {
                    name: "String"
                }
            },
            issuedAtUtc: {
                serializedName: "issuedAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            acceptedAtUtc: {
                serializedName: "acceptedAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            values: {
                serializedName: "values",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            correlationId: {
                serializedName: "correlationId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.ProofAttributeContract = {
    serializedName: "ProofAttributeContract",
    type: {
        name: "Composite",
        className: "ProofAttributeContract",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            },
            attributes: {
                serializedName: "attributes",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            revealed: {
                readOnly: true,
                serializedName: "revealed",
                type: {
                    name: "Boolean"
                }
            },
            selfAttested: {
                serializedName: "selfAttested",
                type: {
                    name: "Boolean"
                }
            },
            conditional: {
                serializedName: "conditional",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.VerificationPolicyRestrictionAttribute = {
    serializedName: "VerificationPolicyRestrictionAttribute",
    type: {
        name: "Composite",
        className: "VerificationPolicyRestrictionAttribute",
        modelProperties: {
            attributeName: {
                serializedName: "attributeName",
                type: {
                    name: "String"
                }
            },
            attributeValue: {
                serializedName: "attributeValue",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.VerificationPolicyRestriction = {
    serializedName: "VerificationPolicyRestriction",
    type: {
        name: "Composite",
        className: "VerificationPolicyRestriction",
        modelProperties: {
            schemaId: {
                serializedName: "schemaId",
                type: {
                    name: "String"
                }
            },
            schemaIssuerDid: {
                serializedName: "schemaIssuerDid",
                type: {
                    name: "String"
                }
            },
            schemaName: {
                serializedName: "schemaName",
                type: {
                    name: "String"
                }
            },
            schemaVersion: {
                serializedName: "schemaVersion",
                type: {
                    name: "String"
                }
            },
            issuerDid: {
                serializedName: "issuerDid",
                type: {
                    name: "String"
                }
            },
            credentialDefinitionId: {
                serializedName: "credentialDefinitionId",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "Composite",
                    className: "VerificationPolicyRestrictionAttribute"
                }
            }
        }
    }
};
exports.VerificationPolicyAttributeContract = {
    serializedName: "VerificationPolicyAttributeContract",
    type: {
        name: "Composite",
        className: "VerificationPolicyAttributeContract",
        modelProperties: {
            policyName: {
                required: true,
                serializedName: "policyName",
                type: {
                    name: "String"
                }
            },
            attributeNames: {
                required: true,
                serializedName: "attributeNames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            restrictions: {
                serializedName: "restrictions",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyRestriction"
                        }
                    }
                }
            }
        }
    }
};
exports.VerificationPolicyPredicateContract = {
    serializedName: "VerificationPolicyPredicateContract",
    type: {
        name: "Composite",
        className: "VerificationPolicyPredicateContract",
        modelProperties: {
            policyName: {
                required: true,
                serializedName: "policyName",
                type: {
                    name: "String"
                }
            },
            attributeName: {
                required: true,
                serializedName: "attributeName",
                type: {
                    name: "String"
                }
            },
            predicateType: {
                required: true,
                serializedName: "predicateType",
                type: {
                    name: "String"
                }
            },
            predicateValue: {
                required: true,
                serializedName: "predicateValue",
                type: {
                    name: "Number"
                }
            },
            restrictions: {
                serializedName: "restrictions",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyRestriction"
                        }
                    }
                }
            }
        }
    }
};
exports.VerificationPolicyRevocationRequirement = {
    serializedName: "VerificationPolicyRevocationRequirement",
    type: {
        name: "Composite",
        className: "VerificationPolicyRevocationRequirement",
        modelProperties: {
            validAt: {
                serializedName: "validAt",
                type: {
                    name: "DateTime"
                }
            },
            validNow: {
                serializedName: "validNow",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.VerificationPolicyParameters = {
    serializedName: "VerificationPolicyParameters",
    type: {
        name: "Composite",
        className: "VerificationPolicyParameters",
        modelProperties: {
            name: {
                required: true,
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            version: {
                required: true,
                serializedName: "version",
                type: {
                    name: "String"
                }
            },
            attributes: {
                serializedName: "attributes",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyAttributeContract"
                        }
                    }
                }
            },
            predicates: {
                serializedName: "predicates",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyPredicateContract"
                        }
                    }
                }
            },
            revocationRequirement: {
                serializedName: "revocationRequirement",
                type: {
                    name: "Composite",
                    className: "VerificationPolicyRevocationRequirement"
                }
            }
        }
    }
};
exports.ProposedAttribute = {
    serializedName: "ProposedAttribute",
    type: {
        name: "Composite",
        className: "ProposedAttribute",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            credentialDefinitionId: {
                serializedName: "credentialDefinitionId",
                type: {
                    name: "String"
                }
            },
            schemaId: {
                serializedName: "schemaId",
                type: {
                    name: "String"
                }
            },
            issuerDid: {
                serializedName: "issuerDid",
                type: {
                    name: "String"
                }
            },
            mimeType: {
                serializedName: "mimeType",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            },
            referent: {
                serializedName: "referent",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.ProposedPredicate = {
    serializedName: "ProposedPredicate",
    type: {
        name: "Composite",
        className: "ProposedPredicate",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            credentialDefinitionId: {
                serializedName: "credentialDefinitionId",
                type: {
                    name: "String"
                }
            },
            issuerDid: {
                serializedName: "issuerDid",
                type: {
                    name: "String"
                }
            },
            schemaId: {
                serializedName: "schemaId",
                type: {
                    name: "String"
                }
            },
            predicate: {
                serializedName: "predicate",
                type: {
                    name: "String"
                }
            },
            threshold: {
                serializedName: "threshold",
                type: {
                    name: "Number"
                }
            },
            referent: {
                serializedName: "referent",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.ProofProposal = {
    serializedName: "ProofProposal",
    type: {
        name: "Composite",
        className: "ProofProposal",
        modelProperties: {
            comment: {
                serializedName: "comment",
                type: {
                    name: "String"
                }
            },
            proposedAttributes: {
                serializedName: "proposedAttributes",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ProposedAttribute"
                        }
                    }
                }
            },
            proposedPredicates: {
                serializedName: "proposedPredicates",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ProposedPredicate"
                        }
                    }
                }
            }
        }
    }
};
exports.VerificationContract = {
    serializedName: "VerificationContract",
    type: {
        name: "Composite",
        className: "VerificationContract",
        modelProperties: {
            connectionId: {
                serializedName: "connectionId",
                type: {
                    name: "String"
                }
            },
            verificationId: {
                serializedName: "verificationId",
                type: {
                    name: "String"
                }
            },
            definitionId: {
                serializedName: "definitionId",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            createdAtUtc: {
                serializedName: "createdAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            updatedAtUtc: {
                serializedName: "updatedAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            isValid: {
                serializedName: "isValid",
                type: {
                    name: "Boolean"
                }
            },
            verifiedAtUtc: {
                serializedName: "verifiedAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            proof: {
                serializedName: "proof",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Composite",
                            className: "ProofAttributeContract"
                        }
                    }
                }
            },
            policy: {
                serializedName: "policy",
                type: {
                    name: "Composite",
                    className: "VerificationPolicyParameters"
                }
            },
            proposal: {
                serializedName: "proposal",
                type: {
                    name: "Composite",
                    className: "ProofProposal"
                }
            },
            verificationRequestData: {
                serializedName: "verificationRequestData",
                type: {
                    name: "String"
                }
            },
            verificationRequestUrl: {
                serializedName: "verificationRequestUrl",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.VerificationPolicyCredentialParameters = {
    serializedName: "VerificationPolicyCredentialParameters",
    type: {
        name: "Composite",
        className: "VerificationPolicyCredentialParameters",
        modelProperties: {
            policyName: {
                serializedName: "policyName",
                type: {
                    name: "String"
                }
            },
            credentialId: {
                serializedName: "credentialId",
                type: {
                    name: "String"
                }
            },
            hidden: {
                serializedName: "hidden",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.VerificationPolicyCredentialContract = {
    serializedName: "VerificationPolicyCredentialContract",
    type: {
        name: "Composite",
        className: "VerificationPolicyCredentialContract",
        modelProperties: {
            policyName: {
                serializedName: "policyName",
                type: {
                    name: "String"
                }
            },
            attributeNames: {
                serializedName: "attributeNames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            availableCredentials: {
                serializedName: "availableCredentials",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CredentialContract"
                        }
                    }
                }
            }
        }
    }
};
exports.VerificationDataPolicyCredentialParameters = {
    serializedName: "VerificationDataPolicyCredentialParameters",
    type: {
        name: "Composite",
        className: "VerificationDataPolicyCredentialParameters",
        modelProperties: {
            verificationData: {
                serializedName: "verificationData",
                type: {
                    name: "String"
                }
            },
            parameters: {
                serializedName: "parameters",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyCredentialParameters"
                        }
                    }
                }
            }
        }
    }
};
exports.CustodianWalletContract = {
    serializedName: "CustodianWalletContract",
    type: {
        name: "Composite",
        className: "CustodianWalletContract",
        modelProperties: {
            walletId: {
                serializedName: "walletId",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.WalletParameters = {
    serializedName: "WalletParameters",
    type: {
        name: "Composite",
        className: "WalletParameters",
        modelProperties: {
            walletId: {
                serializedName: "walletId",
                type: {
                    name: "String"
                }
            },
            ownerName: {
                serializedName: "ownerName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
