// @trinsic/service-clients/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var credentialsServiceClient_1 = require("./credentials/credentialsServiceClient");
exports.CredentialsServiceClient = credentialsServiceClient_1.CredentialsServiceClient;
exports.CredentialsModels = credentialsServiceClient_1.CredentialsServiceModels;
exports.CredentialsMappers = credentialsServiceClient_1.CredentialsServiceMappers;
var walletServiceClient_1 = require("./wallet/walletServiceClient");
exports.WalletServiceClient = walletServiceClient_1.WalletServiceClient;
exports.WalletModels = walletServiceClient_1.WalletServiceModels;
exports.WalletMappers = walletServiceClient_1.WalletServiceMappers;
var providerServiceClient_1 = require("./provider/providerServiceClient");
exports.ProviderServiceClient = providerServiceClient_1.ProviderServiceClient;
exports.ProviderModels = providerServiceClient_1.ProviderServiceModels;
exports.ProviderMappers = providerServiceClient_1.ProviderServiceMappers;
var trinsicClientCredentials_1 = require("./trinsicClientCredentials");
exports.Credentials = trinsicClientCredentials_1.TrinsicClientCredentials;
exports.ProviderCredentials = trinsicClientCredentials_1.TrinsicProviderCredentials;
exports.CredentialsOptions = trinsicClientCredentials_1.TrinsicCredentialsClientOptions;
exports.WalletOptions = trinsicClientCredentials_1.TrinsicWalletClientOptions;
exports.ProviderOptions = trinsicClientCredentials_1.TrinsicProviderClientOptions;
