// @trinsic/service-clients/dist/trinsicClientCredentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TrinsicCredentialsClientOptions = /** @class */ (function () {
    function TrinsicCredentialsClientOptions() {
    }
    return TrinsicCredentialsClientOptions;
}());
exports.TrinsicCredentialsClientOptions = TrinsicCredentialsClientOptions;
var TrinsicWalletClientOptions = /** @class */ (function () {
    function TrinsicWalletClientOptions() {
    }
    return TrinsicWalletClientOptions;
}());
exports.TrinsicWalletClientOptions = TrinsicWalletClientOptions;
var TrinsicProviderClientOptions = /** @class */ (function () {
    function TrinsicProviderClientOptions() {
    }
    return TrinsicProviderClientOptions;
}());
exports.TrinsicProviderClientOptions = TrinsicProviderClientOptions;
var TrinsicClientCredentials = /** @class */ (function () {
    function TrinsicClientCredentials(accessToken) {
        this.accessToken = accessToken;
    }
    TrinsicClientCredentials.prototype.signRequest = function (webResource) {
        webResource.headers.set("Authorization", "Bearer " + this.accessToken);
        return Promise.resolve(webResource);
    };
    return TrinsicClientCredentials;
}());
exports.TrinsicClientCredentials = TrinsicClientCredentials;
var TrinsicProviderCredentials = /** @class */ (function () {
    function TrinsicProviderCredentials(providerKey) {
        this.providerKey = providerKey;
    }
    TrinsicProviderCredentials.prototype.signRequest = function (webResource) {
        webResource.headers.set("Authorization", "Bearer " + this.providerKey);
        return Promise.resolve(webResource);
    };
    return TrinsicProviderCredentials;
}());
exports.TrinsicProviderCredentials = TrinsicProviderCredentials;
