// @trinsic/service-clients/dist/credentials/credentialsServiceClient.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
var msRest = require("@azure/ms-rest-js");
var Models = require("./models");
exports.CredentialsServiceModels = Models;
var Mappers = require("./models/mappers");
exports.CredentialsServiceMappers = Mappers;
var Parameters = require("./models/parameters");
var credentialsServiceClientContext_1 = require("./credentialsServiceClientContext");
exports.CredentialsServiceClientContext = credentialsServiceClientContext_1.CredentialsServiceClientContext;
var CredentialsServiceClient = /** @class */ (function (_super) {
    __extends(CredentialsServiceClient, _super);
    /**
     * Initializes a new instance of the CredentialsServiceClient class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param [options] The parameter options
     */
    function CredentialsServiceClient(credentials, options) {
        return _super.call(this, credentials, options) || this;
    }
    CredentialsServiceClient.prototype.uploadImage = function (uploadedFiles, options, callback) {
        return this.sendOperationRequest({
            uploadedFiles: uploadedFiles,
            options: options
        }, uploadImageOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listNetworks = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listNetworksOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getTransactionAuthorAgreement = function (networkId, options, callback) {
        return this.sendOperationRequest({
            networkId: networkId,
            options: options
        }, getTransactionAuthorAgreementOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.acceptTransactionAuthorAgreement = function (tenantId, options, callback) {
        return this.sendOperationRequest({
            tenantId: tenantId,
            options: options
        }, acceptTransactionAuthorAgreementOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listConnections = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listConnectionsOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createConnection = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createConnectionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getConnection = function (connectionId, options, callback) {
        return this.sendOperationRequest({
            connectionId: connectionId,
            options: options
        }, getConnectionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteConnection = function (connectionId, options, callback) {
        return this.sendOperationRequest({
            connectionId: connectionId,
            options: options
        }, deleteConnectionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listCredentials = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listCredentialsOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createCredential = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createCredentialOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getCredential = function (credentialId, options, callback) {
        return this.sendOperationRequest({
            credentialId: credentialId,
            options: options
        }, getCredentialOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.issueCredential = function (credentialId, options, callback) {
        return this.sendOperationRequest({
            credentialId: credentialId,
            options: options
        }, issueCredentialOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.revokeCredential = function (credentialId, options, callback) {
        return this.sendOperationRequest({
            credentialId: credentialId,
            options: options
        }, revokeCredentialOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteCredential = function (credentialId, options, callback) {
        return this.sendOperationRequest({
            credentialId: credentialId,
            options: options
        }, deleteCredentialOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listVerificationPolicies = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listVerificationPoliciesOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createVerificationPolicy = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createVerificationPolicyOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getVerificationPolicy = function (policyId, options, callback) {
        return this.sendOperationRequest({
            policyId: policyId,
            options: options
        }, getVerificationPolicyOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteVerificationPolicy = function (policyId, options, callback) {
        return this.sendOperationRequest({
            policyId: policyId,
            options: options
        }, deleteVerificationPolicyOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listCredentialDefinitions = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listCredentialDefinitionsOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createCredentialDefinition = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createCredentialDefinitionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getCredentialDefinition = function (definitionId, options, callback) {
        return this.sendOperationRequest({
            definitionId: definitionId,
            options: options
        }, getCredentialDefinitionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createCredentialDefinitionForSchemaId = function (schemaId, body, options, callback) {
        return this.sendOperationRequest({
            schemaId: schemaId,
            body: body,
            options: options
        }, createCredentialDefinitionForSchemaIdOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteCredentialDefinition = function (credentialDefinitionId, options, callback) {
        return this.sendOperationRequest({
            credentialDefinitionId: credentialDefinitionId,
            options: options
        }, deleteCredentialDefinitionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createSchema = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createSchemaOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listSchemas = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listSchemasOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getVerificationDefinition = function (definitionId, options, callback) {
        return this.sendOperationRequest({
            definitionId: definitionId,
            options: options
        }, getVerificationDefinitionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listVerificationDefinitions = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listVerificationDefinitionsOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createVerificationDefinition = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createVerificationDefinitionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteVerificationDefinition = function (verificationDefinitionId, options, callback) {
        return this.sendOperationRequest({
            verificationDefinitionId: verificationDefinitionId,
            options: options
        }, deleteVerificationDefinitionOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.sendMessage = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, sendMessageOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listMessages = function (connectionId, options, callback) {
        return this.sendOperationRequest({
            connectionId: connectionId,
            options: options
        }, listMessagesOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getMessage = function (messageId, options, callback) {
        return this.sendOperationRequest({
            messageId: messageId,
            options: options
        }, getMessageOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listTenants = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listTenantsOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createTenant = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createTenantOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getTenant = function (tenantId, options, callback) {
        return this.sendOperationRequest({
            tenantId: tenantId,
            options: options
        }, getTenantOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteTenant = function (tenantId, options, callback) {
        return this.sendOperationRequest({
            tenantId: tenantId,
            options: options
        }, deleteTenantOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getIssuerStatus = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, getIssuerStatusOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.sendVerificationFromPolicy = function (connectionId, policyId, options, callback) {
        return this.sendOperationRequest({
            connectionId: connectionId,
            policyId: policyId,
            options: options
        }, sendVerificationFromPolicyOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.sendVerificationFromParameters = function (connectionId, body, options, callback) {
        return this.sendOperationRequest({
            connectionId: connectionId,
            body: body,
            options: options
        }, sendVerificationFromParametersOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.sendVerificationFromProposal = function (connectionId, policyId, options, callback) {
        return this.sendOperationRequest({
            connectionId: connectionId,
            policyId: policyId,
            options: options
        }, sendVerificationFromProposalOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createVerificationFromPolicy = function (policyId, options, callback) {
        return this.sendOperationRequest({
            policyId: policyId,
            options: options
        }, createVerificationFromPolicyOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createVerificationFromParameters = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createVerificationFromParametersOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listVerifications = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listVerificationsOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createVerification = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createVerificationOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.getVerification = function (verificationId, options, callback) {
        return this.sendOperationRequest({
            verificationId: verificationId,
            options: options
        }, getVerificationOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.deleteVerification = function (verificationId, options, callback) {
        return this.sendOperationRequest({
            verificationId: verificationId,
            options: options
        }, deleteVerificationOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.verifyVerification = function (verificationId, options, callback) {
        return this.sendOperationRequest({
            verificationId: verificationId,
            options: options
        }, verifyVerificationOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.listWebhooks = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, listWebhooksOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.createWebhook = function (body, options, callback) {
        return this.sendOperationRequest({
            body: body,
            options: options
        }, createWebhookOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.removeWebhook = function (webhookId, options, callback) {
        return this.sendOperationRequest({
            webhookId: webhookId,
            options: options
        }, removeWebhookOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.enableWebhook = function (webhookId, options, callback) {
        return this.sendOperationRequest({
            webhookId: webhookId,
            options: options
        }, enableWebhookOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.disableWebhook = function (webhookId, options, callback) {
        return this.sendOperationRequest({
            webhookId: webhookId,
            options: options
        }, disableWebhookOperationSpec, callback);
    };
    CredentialsServiceClient.prototype.health = function (options, callback) {
        return this.sendOperationRequest({
            options: options
        }, healthOperationSpec, callback);
    };
    return CredentialsServiceClient;
}(credentialsServiceClientContext_1.CredentialsServiceClientContext));
exports.CredentialsServiceClient = CredentialsServiceClient;
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var uploadImageOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/common/upload",
    formDataParameters: [
        Parameters.uploadedFiles,
        Parameters.filename,
        Parameters.contentType
    ],
    contentType: "multipart/form-data",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "String"
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var listNetworksOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/common/networks",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "NetworkContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var getTransactionAuthorAgreementOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/common/networks/{networkId}/txnAuthorAgreement",
    urlParameters: [
        Parameters.networkId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.NetworkTxnAgreementContract
        },
        default: {}
    },
    serializer: serializer
};
var acceptTransactionAuthorAgreementOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/common/networks/{tenantId}/txnAuthorAgreement",
    urlParameters: [
        Parameters.tenantId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var listConnectionsOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/connections",
    queryParameters: [
        Parameters.state
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ConnectionContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createConnectionOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/connections",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.ConnectionInvitationParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.ConnectionContract
        },
        default: {}
    },
    serializer: serializer
};
var getConnectionOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/connections/{connectionId}",
    urlParameters: [
        Parameters.connectionId0
    ],
    responses: {
        200: {
            bodyMapper: Mappers.ConnectionContract
        },
        default: {}
    },
    serializer: serializer
};
var deleteConnectionOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/connections/{connectionId}",
    urlParameters: [
        Parameters.connectionId0
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var listCredentialsOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/credentials",
    queryParameters: [
        Parameters.connectionId1,
        Parameters.state,
        Parameters.definitionId0
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CredentialContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createCredentialOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/credentials",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.CredentialOfferParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.CredentialContract
        },
        default: {}
    },
    serializer: serializer
};
var getCredentialOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/credentials/{credentialId}",
    urlParameters: [
        Parameters.credentialId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.CredentialContract
        },
        default: {}
    },
    serializer: serializer
};
var issueCredentialOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/credentials/{credentialId}",
    urlParameters: [
        Parameters.credentialId
    ],
    requestBody: {
        parameterPath: [
            "options",
            "body"
        ],
        mapper: {
            serializedName: "body",
            type: {
                name: "Dictionary",
                value: {
                    type: {
                        name: "String"
                    }
                }
            }
        }
    },
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var revokeCredentialOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/credentials/{credentialId}",
    urlParameters: [
        Parameters.credentialId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var deleteCredentialOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/credentials/delete/{credentialId}",
    urlParameters: [
        Parameters.credentialId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var listVerificationPoliciesOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/verificationPolicies",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createVerificationPolicyOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/verificationPolicies",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.VerificationPolicyParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.VerificationPolicyContract
        },
        default: {}
    },
    serializer: serializer
};
var getVerificationPolicyOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/verificationPolicies/{policyId}",
    urlParameters: [
        Parameters.policyId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationPolicyContract
        },
        default: {}
    },
    serializer: serializer
};
var deleteVerificationPolicyOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/verificationPolicies/{policyId}",
    urlParameters: [
        Parameters.policyId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var listCredentialDefinitionsOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/definitions/credentials",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CredentialDefinitionContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createCredentialDefinitionOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/definitions/credentials",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.CredentialDefinitionFromSchemaParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.CredentialDefinitionContract
        },
        default: {}
    },
    serializer: serializer
};
var getCredentialDefinitionOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/definitions/credentials/{definitionId}",
    urlParameters: [
        Parameters.definitionId1
    ],
    responses: {
        200: {
            bodyMapper: Mappers.CredentialDefinitionContract
        },
        default: {}
    },
    serializer: serializer
};
var createCredentialDefinitionForSchemaIdOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/definitions/credentials/{schemaId}",
    urlParameters: [
        Parameters.schemaId
    ],
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.CredentialDefinitionParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.CredentialDefinitionContract
        },
        default: {}
    },
    serializer: serializer
};
var deleteCredentialDefinitionOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/definitions/credentials/{credentialDefinitionId}",
    urlParameters: [
        Parameters.credentialDefinitionId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var createSchemaOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/definitions/schemas",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.SchemaParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "String"
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var listSchemasOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/definitions/schemas",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "SchemaRecord"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var getVerificationDefinitionOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/definitions/verifications/{definitionId}",
    urlParameters: [
        Parameters.definitionId1
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationDefinitionContract
        },
        default: {}
    },
    serializer: serializer
};
var listVerificationDefinitionsOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/definitions/verifications",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationDefinitionContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createVerificationDefinitionOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/definitions/verifications",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.ProofRequest, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.VerificationDefinitionContract
        },
        default: {}
    },
    serializer: serializer
};
var deleteVerificationDefinitionOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/definitions/verifications/{verificationDefinitionId}",
    urlParameters: [
        Parameters.verificationDefinitionId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var sendMessageOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/messages",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.BasicMessageParameters, { required: true })
    },
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var listMessagesOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/messages/connection/{connectionId}",
    urlParameters: [
        Parameters.connectionId0
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "BasicMessageRecord"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var getMessageOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/messages/{messageId}",
    urlParameters: [
        Parameters.messageId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.BasicMessageContract
        },
        default: {}
    },
    serializer: serializer
};
var listTenantsOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/tenants",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TenantContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createTenantOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/tenants",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.TenantParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.TenantContract
        },
        default: {}
    },
    serializer: serializer
};
var getTenantOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/tenants/{tenantId}",
    urlParameters: [
        Parameters.tenantId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.TenantContract
        },
        default: {}
    },
    serializer: serializer
};
var deleteTenantOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/tenants/{tenantId}",
    urlParameters: [
        Parameters.tenantId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var getIssuerStatusOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/tenants/issuerStatus",
    responses: {
        200: {
            bodyMapper: Mappers.IssuerStatusContract
        },
        default: {}
    },
    serializer: serializer
};
var sendVerificationFromPolicyOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/verifications/policy/{policyId}/connections/{connectionId}",
    urlParameters: [
        Parameters.connectionId0,
        Parameters.policyId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var sendVerificationFromParametersOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/verifications/policy/connections/{connectionId}",
    urlParameters: [
        Parameters.connectionId0
    ],
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.VerificationPolicyParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var sendVerificationFromProposalOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/verifications/proposal/{policyId}/connections/{connectionId}",
    urlParameters: [
        Parameters.connectionId0,
        Parameters.policyId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var createVerificationFromPolicyOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/verifications/policy/{policyId}",
    urlParameters: [
        Parameters.policyId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var createVerificationFromParametersOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/verifications/policy",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.VerificationPolicyParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var listVerificationsOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/verifications",
    queryParameters: [
        Parameters.connectionId1,
        Parameters.definitionId0
    ],
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createVerificationOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/verifications",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.VerificationParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var getVerificationOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/verifications/{verificationId}",
    urlParameters: [
        Parameters.verificationId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationContract
        },
        default: {}
    },
    serializer: serializer
};
var deleteVerificationOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/verifications/{verificationId}",
    urlParameters: [
        Parameters.verificationId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var verifyVerificationOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/verifications/{verificationId}/verify",
    urlParameters: [
        Parameters.verificationId
    ],
    responses: {
        200: {
            bodyMapper: Mappers.VerificationResult
        },
        default: {}
    },
    serializer: serializer
};
var listWebhooksOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/webhooks",
    responses: {
        200: {
            bodyMapper: {
                serializedName: "parsedResponse",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "WebhookContract"
                        }
                    }
                }
            }
        },
        default: {}
    },
    serializer: serializer
};
var createWebhookOperationSpec = {
    httpMethod: "POST",
    path: "credentials/v1/webhooks",
    requestBody: {
        parameterPath: "body",
        mapper: __assign({}, Mappers.WebhookParameters, { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.WebhookContract
        },
        default: {}
    },
    serializer: serializer
};
var removeWebhookOperationSpec = {
    httpMethod: "DELETE",
    path: "credentials/v1/webhooks/{webhookId}",
    urlParameters: [
        Parameters.webhookId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var enableWebhookOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/webhooks/{webhookId}/enable",
    urlParameters: [
        Parameters.webhookId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var disableWebhookOperationSpec = {
    httpMethod: "PUT",
    path: "credentials/v1/webhooks/{webhookId}/disable",
    urlParameters: [
        Parameters.webhookId
    ],
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
var healthOperationSpec = {
    httpMethod: "GET",
    path: "credentials/v1/health",
    responses: {
        200: {},
        default: {}
    },
    serializer: serializer
};
